src/GafferOSL/OSLShader.cpp

```diff
@@ -761,42 +893,49 @@ bool findSplineParameters( const OSLQuery &query, const OSLQuery::Parameter *par
 		}
 	}
 
-	if( !suffix )
+	if( suffix )
 	{
-		return false;
-	}
+		nameWithoutSuffix = parameter->name.string().substr( 0, parameter->name.string().size() - strlen( suffix ) );
 
-	nameWithoutSuffix = parameter->name.string().substr( 0, parameter->name.string().size() - strlen( suffix ) );
+		positionsParameter = query.getparam( nameWithoutSuffix + "Positions" );
+		if(
+			!positionsParameter ||
+			!positionsParameter->type.is_array() ||
+			positionsParameter->type.basetype != TypeDesc::FLOAT ||
+			positionsParameter->type.aggregate != TypeDesc::SCALAR
+		)
+		{
+			return false;
+		}
 
-	positionsParameter = query.getparam( nameWithoutSuffix + "Positions" );
-	if(
-		!positionsParameter ||
-		!positionsParameter->type.is_array() ||
-		positionsParameter->type.basetype != TypeDesc::FLOAT ||
-		positionsParameter->type.aggregate != TypeDesc::SCALAR
-	)
-	{
-		return false;
-	}
+		valuesParameter = query.getparam( nameWithoutSuffix + "Values" );
+		if(
+			!valuesParameter ||
+			!valuesParameter->type.is_array() ||
+			valuesParameter->type.basetype != TypeDesc::FLOAT ||
+			( valuesParameter->type.aggregate != TypeDesc::SCALAR && valuesParameter->type.vecsemantics != TypeDesc::COLOR )
+		)
+		{
+			return false;
+		}
```

Comment:
May I suggest an approach that I think allows us to avoid modifying this function, and therefore avoids the extra level of indentation, and might be a bit clearer?

- Rename this function to `findGafferSplineParameters()` to denote that it deals with _our_ convention.
- Reimplement `findSplineParameters()` such that it just calls `return findGafferSplineParameters() || find3DelightSplineParameters()`.

Reply:
That sounds good to me, fixed in https://github.com/GafferHQ/gaffer/pull/5432/commits/b768452d86ce5cc472207dd1bb6ac954f7c73918